app/src/main/java/net/justdave/mcstatus/MinecraftServer.java

```diff
@@ -347,41 +354,36 @@ private int readVarInt(InputStream in) throws InputMismatchException {
 		return theInt;
 	}
 
-	public Bitmap getThumbnail(String uri) throws FileNotFoundException,
-			IOException {
+	public Bitmap getThumbnail(String uri) throws IOException {
 		// data:image/png;base64,iVBORw0KGgoAAAANSUhEUgA
 		final String prefixString = "data:image/png;base64,";
-		byte[] imagedata64 = new byte[0];
-		byte[] imagedata = new byte[0];
+		byte[] imageDataBase64;
+		byte[] imageData;
 		if (uri.startsWith(prefixString)) {
-			imagedata64 = uri.substring(prefixString.length()).getBytes();
+			imageDataBase64 = uri.substring(prefixString.length()).getBytes();
 		} else {
 			throw new FileNotFoundException("Not the correct URI Prefix");
 		}
-		imagedata = Base64.decode(imagedata64, Base64.DEFAULT);
+		imageData = Base64.decode(imageDataBase64, Base64.DEFAULT);
 		BitmapFactory.Options onlyBoundsOptions = new BitmapFactory.Options();
 		onlyBoundsOptions.inJustDecodeBounds = true;
-		onlyBoundsOptions.inDither = true;// optional
 		onlyBoundsOptions.inPreferredConfig = Bitmap.Config.ARGB_8888;// optional
-		BitmapFactory.decodeByteArray(imagedata, 0, imagedata.length,
+		BitmapFactory.decodeByteArray(imageData, 0, imageData.length,
 				onlyBoundsOptions);
 		if ((onlyBoundsOptions.outWidth == -1)
 				|| (onlyBoundsOptions.outHeight == -1))
 			return null;
 
-		int originalSize = (onlyBoundsOptions.outHeight > onlyBoundsOptions.outWidth) ? onlyBoundsOptions.outHeight
-				: onlyBoundsOptions.outWidth;
+		int originalSize = Math.max(onlyBoundsOptions.outHeight, onlyBoundsOptions.outWidth);
 
 		double ratio = (originalSize > THUMBNAIL_SIZE) ? (originalSize / THUMBNAIL_SIZE)
 				: 1.0;
 
 		BitmapFactory.Options bitmapOptions = new BitmapFactory.Options();
 		bitmapOptions.inSampleSize = getPowerOfTwoForSampleRatio(ratio);
-		bitmapOptions.inDither = true;// optional
```

Comment:
This was deprecated, not sure what it does though... I figured I could remove it when it's optional.

Reply:
+1